gammapy/data/tests/test_data_store.py

```diff
@@ -227,3 +227,22 @@ def test_datastore_header_info_in_obs_info(data_store):
     assert "GEOLAT" in obs.obs_info
     # make sure we don't add the OBS_INDEX HDUCLAS
     assert "HDUCLAS1" not in obs.obs_info
+
+@requires_data()
+def test_datastore_from_dir_no_obs_index(caplog):
+    """Test the `from_dir` method."""
+    data_store = DataStore.from_dir(
+        "$GAMMAPY_DATA/hess-dl3-dr1/", "hdu-index.fits.gz", "bad_name"
```

Comment:
I think when users provide explicitly a filename that does not exist, it should raise, not warn.

Reply:
You are right. The correct behavior should be:
- if `obs_index_filename = None`  log a warning if the default obs-index is not there
- raise a warning otherwise.